### PR TITLE
tanglescope: fix warnings

### LINKED ISSUE #1247: tanglescope: fix warnings

### PR DESCRIPTION
**Pull request by bingyanglin**
fix tanglescope warnings
fixes #1247

### CODE REVIEW

File: tanglescope/statscollector/stats/noop.hpp [hunk 1]
```diff
@@ -11,12 +11,12 @@ namespace statscollector {
 
 class NoopTXStats : public TXStats {
  public:
-  virtual void trackNewTX(iri::TXMessage&, PrometheusCollector::CountersMap& metrics){};
-  virtual void trackReattachedTX(PrometheusCollector::CountersMap& metrics){};
-  virtual void trackNewBundle(PrometheusCollector::CountersMap& metrics){};
-  virtual void trackConfirmedBundle(int64_t totalValue, uint64_t size, uint64_t avgBundleDuration,
-                                    PrometheusCollector::CountersMap& counters,
-                                    PrometheusCollector::HistogramsMap& histograms){};
+  virtual void trackNewTX(iri::TXMessage&, PrometheusCollector::CountersMap& /* metrics */){};
```
Comment (tsvisabo) [inline tanglescope/statscollector/stats/noop.hpp]: Can you document these functions as we do here for example:
https://github.com/iotaledger/entangled/blob/develop/ciri/core.h

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: tanglescope/statscollector/stats/noop.hpp (hunk 1 above)
Comment (bingyanglin) [reply tanglescope/statscollector/stats/noop.hpp]: Sure, add doc. at https://github.com/iotaledger/entangled/pull/1297/commits/91c84869b89f03062662e03ddc256fbaaf428bd6

File: tanglescope/blowballcollector.cpp [hunk 2]
```diff
@@ -57,7 +58,10 @@ void BlowballCollector::analyzeBlowballsPeriodically() {
   auto& thisRef = *this;
   if (_snapshotInterval > 0) {
     pubWorker.schedule_periodically(pubThread.now(), std::chrono::seconds(_snapshotInterval),
-                                    [&thisRef, buckets](auto scbl) { thisRef.analyzeBlowballs(buckets); });
+                                    [&thisRef, buckets](auto scbl) {
```
Comment (thibault-martinez) [inline tanglescope/blowballcollector.cpp]: Unlike in C, in C++ you can just omit the variable name if you don't use it.
So you can do `[&thisRef, buckets](auto) {` instead and not have to use the `UNUSED` macro.
Same goes for all the following same cases.

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: tanglescope/statscollector/stats/noop.hpp (hunk 1 above)
Comment (oopsmonk) [reply tanglescope/statscollector/stats/noop.hpp]: please add directions in parameters as well, see https://github.com/iotaledger/entangled/blob/develop/ciri/api/api.h

**Review by thibault-martinez — DISMISSED** [review]

File: tanglescope/statscollector/stats/noop.hpp (hunk 1 above)
Comment (bingyanglin) [reply tanglescope/statscollector/stats/noop.hpp]: Thank you for pointing this out. Modified at https://github.com/iotaledger/entangled/pull/1297/commits/092494dbcb4156b95d4c4ddbb22b6ff058dd5a44

File: tanglescope/blowballcollector.cpp (hunk 2 above)
Comment (bingyanglin) [reply tanglescope/blowballcollector.cpp]: Got it. Modified all of them at https://github.com/iotaledger/entangled/pull/1297/commits/092494dbcb4156b95d4c4ddbb22b6ff058dd5a44

File: tanglescope/statscollector/stats/noop.hpp (hunk 1 above)
Comment (bingyanglin) [reply tanglescope/statscollector/stats/noop.hpp]: Modified the func. dir. at https://github.com/iotaledger/entangled/pull/1297/commits/3beeafb0071d77cae0153e14a821a72fbc1b9aa7

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**